nucypher/blockchain/eth/actors.py

```diff
@@ -502,10 +543,18 @@ def initialize_stake(self,
 
         amount = NU(int(amount), 'NuNit')
 
+        current_period = self.miner_agent.get_current_period()
+        stake = Stake(owner_address=self.checksum_public_address,
+                      index=len(self.stakes)+1,
+                      start_period=current_period,
+                      end_period=current_period + lock_periods,
+                      value=amount,
+                      economics=self.economics)
+
         staking_transactions = OrderedDict()  # type: OrderedDict # Time series of txhases
 
         # Validate
-        assert self.__validate_stake(amount=amount, lock_periods=lock_periods)
+        assert self.__validate_stake(stake=stake)
```

Comment:
This validates stake itself, also could be situation when deposit of new stake increases amount of locked tokens more than allowed

Reply:
:heavy_check_mark: